src/tests/Microsoft.PowerFx.Interpreter.Tests/Scenarios/ScenarioSymbolTables.cs

```diff
@@ -0,0 +1,192 @@
+            }
+
+            var checkResult = engine.Check(parseResult, options: null, symbolTable);
+            var resultValue = checkResult.GetEvaluator().Eval() as ObjectRecordValue;
+
+            Assert.NotNull(resultValue);
+            Assert.IsType(expectedType, resultValue.Source);
+        }
+
+        private class BaseRecord
+        {
+            public string TableName { get; set; }
+
+            public long RecordId { get; set; }
+
+            public static BaseRecord CreateRecord(string tableName)
+            {
+                BaseRecord record = null;
+
+                switch (tableName)
+                {
+                    case "Customer":
+                        record = new CustomerTable();
+                        break;
+                    case "Vendor":
+                        record = new VendorTable();
+                        break;
+                }
+
+                return record;
+            }
+    }
+
+        private class CustomerTable : BaseRecord
+        {
+            public string CustomerId { get; set; }
+        }
+
+        private class VendorTable : BaseRecord
+        {
+            public string VendorId { get; set; }
+        }
+
+        private class FindRecord : ReflectionFunction
+        {
+            private readonly TypeMarshallerCache _typeCache;
+
+            public FindRecord(TypeMarshallerCache cache, RecordType recordType)
+                : base("FindRecord", recordType, FormulaType.String)
+            {
+                _typeCache = cache;
+            }
+
+            public FindRecord(TypeMarshallerCache cache)
+                : this(cache, RecordType.Empty())
+            {
+            }
+
+            public RecordValue Execute(StringValue tableName)
+            {
+                var record = BaseRecord.CreateRecord(tableName.Value);
+
+                return record != null ? _typeCache.Marshal(record, record.GetType()) as RecordValue : RecordValue.Empty();
+            }
+        }
+
+        private class FindRecordVisitor : TexlVisitor
+        {
+            private readonly List<string> _recordTypes = new List<string>();
+
+            public IEnumerable<string> RecordTypes => _recordTypes;
+
+            public override void PostVisit(StrInterpNode node)
+            {
+            }
+
+            public override void PostVisit(DottedNameNode node)
+            {
+            }
+
+            public override void PostVisit(UnaryOpNode node)
+            {
+            }
+
+            public override void PostVisit(BinaryOpNode node)
+            {
+            }
+
+            public override void PostVisit(VariadicOpNode node)
+            {
+            }
+
+            public override void PostVisit(CallNode node)
+            {
+                if (node.Head.Name.Equals("FindRecord"))
+                {
+                    _recordTypes.Add((node.Args.ChildNodes.First() as StrLitNode).Value);
```

Comment:
This could be a null ref if the first arg is not a string literal.

Reply:
Yes indeed. As you mentioned, there's no way to set the type for the reflection function arg is a literal, only FormulaType.String...

Reply:
Right, at the very least, we should check and just return, instead of throwing a nullref, though.